internal/controllers/rewards.go

```diff
@@ -174,13 +174,11 @@ func (r *RewardsController) GetUserRewards(c *fiber.Ctx) error {
 
 		vehicleMinted := device.TokenId != nil
 
-		var vehicleIntegsWithSignals []string
+		integSignalsThisWeek := set.New[string]()
 		if device.TokenId != nil {
-			vehicleIntegsWithSignals = integrationsByTokenID[*device.TokenId]
+			integSignalsThisWeek = integrationsByTokenID[*device.TokenId]
```

Comment:
If `*device.TokenId` is not a key in `integrationsByTokenID` then we're gonna have problems maybe?